public/javascripts/Progress/src/MistakeCarousel.js

```diff
@@ -13,12 +13,30 @@ function MistakeCarousel() {
         let labelTypesWithoutData = [];
         labelTypes.forEach((l) => (data[l].length > 0 ? labelTypesWithData : labelTypesWithoutData).push(l));
 
-        // Add subheader for labeling mistakes section, listing label types without validations.
-        let mistakesSubheader = document.getElementById('mistakes-subheader');
+        // This is the list of label types without validations.
         const translatedTypes = labelTypesWithoutData.map((l) => i18next.t(`common:${util.camelToKebab(l)}`));
-        mistakesSubheader.textContent = i18next.t('mistakes-subheader', { labelTypes: translatedTypes });
+
+        // Get state of user (has no mistakes or has a mistake labeling).
+        let userState = '';
+        // User has no mistakes with their labels.
+        if (labelTypesWithoutData.length === labelTypes.length) {
+            userState = 'no-mistakes-subheader'
+        // Otherwise user has at least one mistake with their labels.
+        } else {
+            userState = 'mistakes-subheader'
+        }
+
+        let mistakesSubheader = document.getElementById('mistakes-subheader-display');
+        mistakesSubheader.innerHTML = i18next.t(userState)
+
+        // If the user has made a mistake, and they have not made a mistake of all types
+        // append the text regarding what label types they have not made a mistake on.
+        if (userState === 'mistakes-subheader' && labelTypesWithData.length !== labelTypes.length) {
+            mistakesSubheader.textContent += " " + i18next.t('mistakes-info', { labelTypes: translatedTypes });
```

Comment:
Let's build the whole string first, and then set the innerHTML of the subheader one time! Both for performance, and for simplicity :) Glad you got the link working in the translations file!

Reply:
Hmm fair but adding `mistakesSubheader.innerHTML = i18next.t(userState)` after the if statement removes the list of labels where the user has not made a mistake on.

Reply:
Should be easy to fix!